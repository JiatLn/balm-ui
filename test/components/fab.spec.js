import { shallowMount } from '@vue/test-utils';
import UiFab from '@/components/fab/fab.vue';

describe('<ui-fab>', () => {
  it('renders default fab', () => {
    const wrapper = shallowMount(UiFab);

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.element.tagName).toBe('BUTTON');
    expect(wrapper.classes()).toContain('mdc-fab');
  });

  it('renders extended fab', () => {
    const wrapper = shallowMount(UiFab, {
      propsData: {
        extended: true
      }
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.classes()).toContain('mdc-fab--extended');
  });
});
