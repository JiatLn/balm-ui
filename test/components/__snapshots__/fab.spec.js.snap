// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ui-fab> renders default fab 1`] = `
<button
  class="mdc-fab"
  type="button"
>
  <div
    class="mdc-fab__ripple"
  />
   
  <!---->
</button>
`;

exports[`<ui-fab> renders extended fab 1`] = `
<button
  class="mdc-fab mdc-fab--extended"
  type="button"
>
  <div
    class="mdc-fab__ripple"
  />
   
  <!---->
   
  <span
    class="mdc-fab__label"
  />
   
</button>
`;
